### DISCUSSION TITLE
`root.tsx` vs `routes/index.tsx` for inserting a custom Chat or Modal widget

### DISCUSSION THREAD

**Discussion by cliffordfajardo** [remix-run/remix#5891] [Q&A]
**Scenario**
I have a custom chat modal I'm planning on building for an app;
<img src="https://user-images.githubusercontent.com/6743796/227391450-87544b44-f2d6-4826-b026-9007698638fb.png"  height="300"/>

Should I insert it at `root.tsx` or `routes/index.tsx`?

**root.tsx**
```tsx
  return (
    <html lang="en">
      <head>
        <Meta />
        <Links />
      </head>
      <body>
        <ScrollRestoration />
        <Scripts />
        <ChatWidget /> <⬅️--------------------------
      </body>
    </html>
  );
}
```

**Comment by cliffordfajardo** [comment]
Upon writing this question I realized a few things
The answer of whether `root.tsx` or `routes/index.tsx` depends on
-  whether I want my <ChatWidget /> to be accessible on all pages in my app 
- or just accessible on 1 page in my app 

If I place it in `routes/index.tsx`
when I navigate away from that route, that widget would be gone, whereas, if I defined it in `root.tsx` it would still be stick around as I browser across different paths in my app

**Reply by kiliman** [reply]
You can always add it to a parent layout depending on what part of your site that widget should be located. This will make it available to all child routes.
